src/browser-frontend/views/browser/chrome/page.jsx

```diff
@@ -38,6 +44,10 @@ export default class Page extends PureComponent {
     this.props.dispatch(WebContentsActions.events.pageDidMount({ pageId: this.props.pageId }));
   }
 
+  componentDidUpdate() {
+    this.props.dispatch(WebContentsActions.events.pageDidChangeImpl({ pageId: this.props.pageId }));
```

Comment:
Seems a little misleading to always say `pageDidChangeImpl` here.  It seems most updates will not result in an impl change - can we check to see if props.implType has actually changed before firing?